### PR TITLE
#167899328 Setup Article Update

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?
This PR enables a user to update his existing article

#### Description of Task proposed in this pull request?
-the method updateArticle was added to the Articles controller
-Validation for the update article feature
-Article update routes were added
-swagger docs was added

#### How should this be manually tested (Quality Assurance)?

```
- npm run start:dev (starting development server)
-create an article localhost:9000/api/v1/articles/create
-update the article
-localhost:9000/api/v1/articles/update?slug=hellobabye

```

#### What are the relevant pivotal tracker stories?
- [167899328](https://www.pivotaltracker.com/story/show/167899328)

#### What I have learned working on this feature:
- I learnt more on  how to use sequelize 
- I learnt about various front end editors that can be incoporated into this task

#### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
Screenshot 2019-08-16 at 1 52 30 PM

### CODE REVIEW

File: test/articles/update.test.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        .patch(`${BASE_URL}/articles/update?slug=${slug2}`)
+        .set('Authorization', userToken)
+        .send(ArticleData4);
+
+      expect(response.status).to.equal(200);
+      expect(response.body.publishedAt).to.equal(null);
+    });
+  });
+  context(
+    'when a registered enter valid details for update of article he didnt create',
```
Comment (henryade) [inline test/articles/update.test.js]: I believe you missed 'user' in the context of this test.

Comment (Adekoreday) [reply test/articles/update.test.js]: > I believe you missed 'user' in the context of this test.

yes i did fixed thanks

**Review by henryade — DISMISSED** [review]

File: server/controllers/Articles.js [hunk 2]
```diff
@@ -76,5 +76,44 @@ class Articles {
...
+  /**
+   * @name updateArticle
+   * @async
+   * @static
+   * @memberof Articles
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new user
+   */
+  static async updateArticle(req, res) {
```
Comment (chialuka) [inline server/controllers/Articles.js]: Please change the method's name to update

**Review by chialuka — COMMENTED** [review]
Good job with the image upload mock

File: server/controllers/Articles.js [hunk 3]
```diff
@@ -76,5 +76,44 @@ class Articles {
...
+    try {
+      let image;
+      const {
```
Comment (dinobi) [inline server/controllers/Articles.js:372]: why do you always do this new lines?

File: test/articles/update.test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+          .request(app)
+          .patch(`${BASE_URL}/articles/update/${slug1}`)
+          .set('Authorization', userToken2)
+          .send(ArticleData2);
+        expect(response.status).to.equal(404);
+      });
+    }
+  );
+  context('when there is an internal error', () => {
+    it('will throw error', async () => {
```
Comment (dinobi) [inline test/articles/update.test.js]: "returns a server error"

File: test/articles/update.test.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        .patch(`${BASE_URL}/articles/update/${slug2}`)
+        .set('Authorization', userToken)
+        .send(ArticleData4);
+
+      expect(response.status).to.equal(200);
+      expect(response.body.publishedAt).to.equal(null);
+    });
+  });
+  context(
+    'when a registered enter valid details for update of article he didnt create',
```
Comment (dinobi) [inline test/articles/update.test.js]: "when a user attempts to update another users article"

File: test/articles/update.test.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+          .request(app)
+          .patch(`${BASE_URL}/articles/update/${'unknown'}`)
+          .set('Authorization', userToken)
+          .send(ArticleData2);
+        expect(response.status).to.equal(404);
+        expect(response.body.message).to.equal('Article not found');
+      });
+    }
+  );
+  context('when a registered user enter valid details', () => {
```
Comment (dinobi) [inline test/articles/update.test.js]: Is this spec not a repetition of the one on `112`

File: test/articles/update.test.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+          .set('Authorization', userToken)
+          .send(ArticleData4);
+        slug1 = response.body.slug;
+        expect(response.status).to.equal(200);
+        expect(response.body.publishedAt).to.equal(null);
+      });
+    }
+  );
+  context(
+    `when a registered user  enter article with
```
Comment (dinobi) [inline test/articles/update.test.js]: check multispace after user

File: server/controllers/Articles.js (hunk 3 above)
Comment (Adekoreday) [reply server/controllers/Articles.js:372]: husky is formatting them by default on every commit

File: test/articles/update.test.js (hunk 5 above)
Comment (Adekoreday) [reply test/articles/update.test.js]: done thanks

File: test/articles/update.test.js (hunk 7 above)
Comment (Adekoreday) [reply test/articles/update.test.js]: done thanks

File: test/articles/update.test.js (hunk 6 above)
Comment (Adekoreday) [reply test/articles/update.test.js]: fixed

File: test/articles/update.test.js (hunk 7 above)
Comment (Adekoreday) [reply test/articles/update.test.js]: done thanks

**Review by dinobi — APPROVED** [review]

**Pull request merged**